packages/core/src/odata-common/selectable/edm-field.ts

```diff
@@ -0,0 +1,163 @@
+import BigNumber from 'bignumber.js';
+import moment from 'moment';
+import { EdmTypeShared } from '../edm-types';
+import { Entity } from '../entity';
+import { Filter } from '../filter';
+import { Time } from '../time';
+import { getEntityConstructor } from './complex-type-field';
+import { ConstructorOrField } from './constructor-or-field';
+import { Field, FieldType } from './field';
+
+export type SortableEdmType =
+  | 'Edm.Decimal'
+  | 'Edm.Double'
+  | 'Edm.Single'
+  | 'Edm.Float'
+  | 'Edm.Int16'
+  | 'Edm.Int32'
+  | 'Edm.Int64'
+  | 'Edm.SByte'
+  | 'Edm.Byte'
+  | 'Edm.DateTime'
+  | 'Edm.DateTimeOffset'
+  | 'Edm.Time'
+  | 'Edm.Date'
+  | 'Edm.Duration'
+  | 'Edm.TimeOfDay';
+
+export function isSortableEdmType(edmType: EdmTypeShared<'any'>): boolean {
+  return [
+    'Edm.Boolean',
+    'Edm.Decimal',
+    'Edm.Double',
+    'Edm.Single',
+    'Edm.Float',
+    'Edm.Int16',
+    'Edm.Int32',
+    'Edm.Int64',
+    'Edm.SByte',
+    'Edm.Byte',
+    'Edm.DateTime',
+    'Edm.Time',
+    'Edm.Date',
+    'Edm.Duration',
+    'Edm.TimeOfDay'
+  ].includes(edmType);
+}
+
+type NullableFieldType<
+  FieldT extends FieldType,
+  NullableT extends boolean
+> = NullableT extends true ? FieldT | null : FieldT;
+
+export type NonNullableFieldTypeByEdmType<
+  EdmOrFieldT extends EdmTypeShared<'any'> | FieldType
+> = EdmOrFieldT extends 'Edm.String'
+  ? string
+  : EdmOrFieldT extends 'Edm.Boolean'
+  ? boolean
+  : EdmOrFieldT extends 'Edm.Decimal'
+  ? BigNumber
+  : EdmOrFieldT extends 'Edm.Double'
+  ? number
+  : EdmOrFieldT extends 'Edm.Single'
+  ? number
+  : EdmOrFieldT extends 'Edm.Float'
+  ? number
+  : EdmOrFieldT extends 'Edm.Int16'
+  ? number
+  : EdmOrFieldT extends 'Edm.Int32'
+  ? number
+  : EdmOrFieldT extends 'Edm.Int64'
+  ? BigNumber
+  : EdmOrFieldT extends 'Edm.SByte'
+  ? number
+  : EdmOrFieldT extends 'Edm.Binary'
+  ? string
+  : EdmOrFieldT extends 'Edm.Guid'
+  ? string
+  : EdmOrFieldT extends 'Edm.Byte'
+  ? number
+  : EdmOrFieldT extends 'Edm.DateTime'
+  ? moment.Moment
+  : EdmOrFieldT extends 'Edm.Time'
+  ? Time
+  : EdmOrFieldT extends 'Edm.Date'
+  ? moment.Moment
+  : EdmOrFieldT extends 'Edm.Duration'
+  ? moment.Duration
+  : EdmOrFieldT extends 'Edm.TimeOfDay'
+  ? Time
+  : EdmOrFieldT extends 'Edm.Enum'
+  ? string
+  : EdmOrFieldT extends 'Edm.Any'
+  ? any
+  : EdmOrFieldT extends FieldType
+  ? EdmOrFieldT
+  : never;
+
+export type FieldTypeByEdmType<
+  EdmT extends EdmTypeShared<'any'>,
+  NullableT extends boolean
+> = NullableFieldType<NonNullableFieldTypeByEdmType<EdmT>, NullableT>;
+
+/**
+ * Represents a property of an OData entity with an Edm type.
+ *
+ * `EdmTypeField`s are used as static properties of entities and are generated from the metadata, i.e. for each property of
```

Comment:
perhaps avoid the term metadata -> service definitions, edmx files just to avoid confusion with the the sdk metadata WRT APIhub

Reply:
Added OData notion